apps/src/code-studio/initApp/project.js

```diff
@@ -586,22 +586,40 @@ var projects = (module.exports = {
       this.updateChannels_(callback);
     }
   },
-  setLibraryDetails(newName, newDescription, publishing = undefined) {
+  /**
+   * Updates the current channel's library details.
+   *
+   * @param {Object} config - Object containing library details.
+   * @param {string} config.newName
+   * @param {string} config.newDescription
+   * @param {string} config.newVersionId - S3 version ID for the current library version. Passing this value as -1 will nullify libraryLatestVersion.
+   * @param {boolean} config.publishing - true if library is being published, false if library is being unpublished, undefined otherwise.
+   */
+  setLibraryDetails(config = {}) {
     current = current || {};
+    const {newName, newDescription, newVersionId, publishing} = config;
+
     if (
       current.libraryName !== newName ||
       current.libraryDescription !== newDescription ||
+      current.latestLibraryVersion !== newVersionId ||
       publishing !== undefined
     ) {
       current.libraryName = newName;
       current.libraryDescription = newDescription;
 
+      if (newVersionId) {
+        current.latestLibraryVersion =
+          newVersionId === -1 ? null : newVersionId;
+      }
+
       if (publishing) {
         // Tells the server to set libraryPublishedAt timestamp.
         current.publishLibrary = true;
```

Comment:
Why do we have the server set the timestamp rather than passing it as part of this call?

Reply:
that's just a pattern i've learned in the past -- always let the server set its own timestamps so that all times and their formats are consistent

Reply:
👍